app/models/accounting/transaction.rb

```diff
@@ -52,6 +57,14 @@ def self.find_or_create_from_qb_object(transaction_type:, qb_object:)
     transaction.save!(validate: false)
   end
 
+  def self.by_date
+    order(txn_date: :asc)
+  end
+
+  def self.previous(loan_id)
```

Comment:
Once it's an instance method, it also shouldn't need to take an argument, since you will have the `project_id` of self.